network/p2p/capabilities.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2019-2024 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package p2p
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	"github.com/libp2p/go-libp2p/core/discovery"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	libpeerstore "github.com/libp2p/go-libp2p/core/peerstore"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/logging"
+	algoDht "github.com/algorand/go-algorand/network/p2p/dht"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// Capability represents functions that some nodes may provide and other nodes would want to know about
+type Capability string
+
+const (
+	// Archival nodes
+	Archival Capability = "archival"
+	// Catchpoints storing nodes
+	Catchpoints = "catchpointStoring"
+	// Gossip nodes are non permissioned relays
+	Gossip = "gossip"
+)
+
+const operationTimeout = time.Second * 5
+const maxAdvertisementInterval = time.Hour * 22
```

Comment:
How were these constant values selected?

Reply:
I do not see any comments about it in the [originating PR](https://github.com/algorand/go-algorand/pull/5702/files#diff-b5831e1af03f2c716ed89ae9eef06583bcef2f1931ec23f7fd69b44150e43b45) but I remember seeing 22h in libp2p's kad-dht [spec](https://github.com/libp2p/specs/blob/master/kad-dht/README.md)